packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts

```diff
@@ -0,0 +1,279 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { strict as assert } from "assert";
+import * as fs from "fs";
+import random from "random-js";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { SegmentGroup } from "../mergeTreeNodes";
+import { IMergeTreeOp, MergeTreeDeltaType } from "../ops";
+import {
+    annotateRange,
+    applyMessages,
+    doOverRange,
+    IConfigRange,
+    IMergeTreeOperationRunnerConfig,
+    insertAtRefPos,
+    removeRange,
+    ReplayGroup,
+    replayResultsPath,
+    TestOperation,
+    generateClientNames,
+} from "./mergeTreeOperationRunner";
+import { TestClient } from "./testClient";
+import { TestClientLogger } from "./testClientLogger";
+
+interface IRollbackFarmConfig extends IMergeTreeOperationRunnerConfig {
+    minLength: IConfigRange;
+    rollbackOnlyClients: IConfigRange;
+    editOnlyClients: IConfigRange;
+    rollbackAndEditClients: IConfigRange;
+    opsPerRollbackRange: IConfigRange;
+}
+
+interface IClients {
+    clients: TestClient[];
+    rollbackOnlyCount: number;
+    editOnlyCount: number;
+    rollbackAndEditCount: number;
+}
+
+const allOperations: TestOperation[] = [
+    removeRange,
+    annotateRange,
+    insertAtRefPos,
+];
+
+export const debugOptions: IRollbackFarmConfig = {
+    minLength: { min: 2, max: 2 },
+    rollbackOnlyClients: { min: 0, max: 0 },
+    editOnlyClients: { min: 0, max: 0 },
+    rollbackAndEditClients: { min: 1, max: 1 },
+    opsPerRollbackRange: { min: 1, max: 15 },
+    opsPerRoundRange: { min: 1, max: 100 },
+    rounds: 5,
+    operations: allOperations,
+    incrementalLog: true,
+    growthFunc: (input: number) => input + 1,
+};
+
+export const defaultOptions: IRollbackFarmConfig = {
+    minLength: { min: 1, max: 512 },
```

Comment:
no need to go so big. generally, the most interesting cases are small length, so lots of conflicts on position, i'd probably make max 32 for something like this. big enough to get deepish trees, but small enough to run fast. we never fine bugs in the high length of conflict farm, so it more there for sanity